### ISSUE TITLE
[v2] gatsby build hangs on WSL Ubuntu

### ISSUE THREAD

**Issue by dustinheestand** [gatsbyjs/gatsby#6540]
## Description

Build hangs on "Building production Javascript and CSS bundles," after "bootstrap finished." This doesn't happen on Gatsby v1, or with `gatsby develop`.

### Steps to reproduce

1. Make a new gatsby site using command: `gatsby new tutorial-part-one https://github.com/gatsbyjs/gatsby-starter-hello-world#v2`. 
2. Run `gatsby build`.

### Expected result

Gatsby should build the production site.

### Actual result

Gatsby hangs on "Building production Javascript and CSS bundles." I let it run for 10 minutes and then quit. This occurs on both the gatsby-cli 

### Environment

- System:
  - OS: Linux 4.4 Ubuntu 16.04.4 LTS (Xenial Xerus) (wsl on Windows build 17692)
  - CPU: x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  - Shell: 5.1.1 - /usr/bin/zsh
- Binaries:
  - Node: 8.11.3 - /usr/local/bin/node
  - npm: 6.2.0 - /usr/local/bin/npm
- npmPackages:
  - gatsby: next => 2.0.0-beta.46
  - gatsby-plugin-react-helmet: next => 3.0.0-beta.3
- npmGlobalPackages:
  - gatsby-cli: 2.0.0-beta6
  - also occurs on -gatsby-cli: 1.1.58

### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

**Comment by dustinheestand** [comment]
I just had a try - didn't work. However I did manage to get `gatsby build` to work on the beta on a different Ubuntu machine. That machine is running Ubuntu 18.04 but otherwise has the same config.

Is there any way for me to pinpoint this issue better? The `--verbose` flag doesn't give me any new info.

**Comment by KyleAMathews** [comment]
Node inspector could help https://nodejs.org/en/docs/guides/debugging-getting-started/

**Comment by azangru** [comment]
Uh-oh, now it started working fine again. I swear it consistently hung after a fresh install of the latest version of 2.0.0-beta.47. I removed node_modules and reinstalled dependencies several times, and it continued to freeze. And then, after one last reinstall, it started working again. Very strange.

**Comment by m-allanson** [comment]
@pieh reported some oddities with WSL and a recent Windows update. Maybe this is the same thing?

**Comment by piotrkwiecinski** [comment]
I've just checked the starter on my machine:
```
  System:
    OS: Linux 4.15 Ubuntu 18.04 LTS (Bionic Beaver)
    CPU: x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 4.4.19 - /bin/bash
  Binaries:
    Node: 8.11.3 - /usr/bin/node
    Yarn: 1.7.0 - /usr/bin/yarn
    npm: 5.6.0 - /usr/bin/npm
  Browsers:
    Chrome: 67.0.3396.99
    Firefox: 61.0.1
  npmPackages:
    gatsby: next => 2.0.0-beta.48 
```

```
gatsby build
success open and validate gatsby-config — 0.007 s
success load plugins — 0.081 s
success onPreInit — 0.021 s
success delete html and css files from previous builds — 0.007 s
success initialize cache — 0.004 s
success copy gatsby files — 0.007 s
success onPreBootstrap — 0.000 s
success source and transform nodes — 0.009 s
success building schema — 0.057 s
success createPages — 0.000 s
success createPagesStatefully — 0.013 s
success onPreExtractQueries — 0.000 s
success update schema — 0.037 s
success extract queries from components — 0.010 s
success run graphql queries — 0.008 s — 1/1 153.93 queries/second
success write out page data — 0.002 s
success write out redirect data — 0.000 s
success onPostBootstrap — 0.000 s

info bootstrap finished - 1.647 s

success Building production JavaScript and CSS bundles — 7.605 s
success Building static HTML for pages — 0.731 s — 1/1 11.18 pages/second
info Done building in 9.985 sec
```
Seams to work.

**Comment by azangru** [comment]
Right, got the issue (not sure whether it’s quite the same issue as reported in the original message) reliably reproduce on my machine:

Steps:

1) use gatsby v.2.0.0-beta.42
2) run gatsby build
3) build completes successfully

...

4) update gatsby to v.2.0.0-beta.47
5) run gatsby build
6) build completes successfully

...

7) remain on gatsby to v.2.0.0-beta.47
8) delete the `public` folder
9) run gatsby build
10) observe that the build hangs at the step`Building static HTML for pages`

===

Running ubuntu 16.04, node v8.9.4, npm v.6.2.0.

**Comment by sorja** [comment]
I am on (legacy) WSL and I run also in the following problem: 
`gatsby build` (or `yarn build`) hangs on `⠐ Building production JavaScript and CSS bundles`
`gatsby@next` and works on macos

**Comment by awidjaja** [comment]
Building production JavaScript and CSS bundles never completed.

I am on Ubuntu WSL too

Distributor ID: Ubuntu
Description:    Ubuntu 18.04 LTS
Release:        18.04
Codename:       bionic

Gatsby version 2.0.0-beta.7
Yarn 1.7.0
Node v10.6.0

**Comment by joewood** [comment]
Anyone found a workaround for this yet? Or insight into which stage is causing the problem?

**Comment by HriBB** [comment]
Check your memory
https://github.com/gatsbyjs/gatsby/issues/7013#issuecomment-416764877

**Comment by KyleAMathews** [comment]
👍 closing this. Please open a new issue if there's something we can do to help make using Gatsby smooth on WSL Ubuntu!

**Comment by LubomirGeorgiev** [comment]
This issue has been fixed in #12636
